src/Symfony/Component/DependencyInjection/Compiler/ReplaceAliasByActualDefinitionPass.php

```diff
@@ -131,6 +131,6 @@ private function updateFactoryServiceReference($factoryService, $currentId, $new
             return;
         }
 
-        return $currentId === $factoryService ? $newId : $currentId;
+        return $currentId === $factoryService ? $newId : $factoryService;
```

Comment:
`$currentId` is the current ID of the removed alias, while `$factoryService` is the original ID of the factory service.

If the factory service equals the ID of the removed alias, it should be updated to become the actual service ID. However, if the factory service does not equal the ID of the removed alias, the original value should be used.

With the old code, it was always replacing the original code, either with the actual definition ID or the ID of the removed alias (which was causing the troubles people were reporting).